tests/integration/golang/aim/metric/search_test.go

```diff
@@ -37,102 +40,6056 @@ func (s *SearchMetricsTestSuite) Test_Ok() {
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric2Run2,
+			},
+		},
+		// Search Metric Last Step(less or equals operation) and Run Duration
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration == 333333)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndNotEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration != 333333)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndGreater",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration > 0)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndGreaterOrEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration >= 0)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndLess",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration < 333333)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunDurationOperationLessOrEqualsAndLessOrEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step  <= 3) and run.duration <= 333333)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		// Search Metric Last Step (equal operation) and Run Hash
```

Comment:
some of these cases have a description comment but manuy do not -- i'd say get rid of the comment and let name speak for itself

Reply:
Yep I put them on the different sections, but I think the name is clear as well